File: app/src/main/java/com/stevenschoen/putionew/fragments/AccountFragment.java
package com.stevenschoen.putionew.fragments;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.stevenschoen.putionew.PutioApplication;
import com.stevenschoen.putionew.PutioUtils;
import com.stevenschoen.putionew.R;
import com.stevenschoen.putionew.model.account.PutioAccountInfo;
import com.stevenschoen.putionew.model.responses.AccountInfoResponse;
import com.trello.rxlifecycle2.components.support.RxFragment;

import io.reactivex.android.schedulers.AndroidSchedulers;
import io.reactivex.annotations.NonNull;
import io.reactivex.functions.Consumer;

public class AccountFragment extends RxFragment {

	private PutioUtils utils;

	private TextView textDiskTotal;
	private TextView textDiskFree;
	private TextView textEmail;
	private TextView textName;
	
	@Override
	public void onCreate(Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);

		this.utils = ((PutioApplication) getActivity().getApplication()).getPutioUtils();
	}
	
	@Override
	public View onCreateView(LayoutInflater inflater, ViewGroup container,
			Bundle savedInstanceState) {
		final View view = inflater.inflate(R.layout.account, container, false);
		
		textName = view.findViewById(R.id.text_behind_accountname);
		textEmail = view.findViewById(R.id.text_behind_accountemail);
		textDiskFree = view.findViewById(R.id.text_storage_amountfree);
		textDiskTotal = view.findViewById(R.id.text_storage_amounttotal);
		
		invalidateAccountInfo();
		
		return view;
	}

	public void invalidateAccountInfo() {
		utils.getRestInterface().account()
				.compose(this.<AccountInfoResponse>bindToLifecycle())
				.observeOn(AndroidSchedulers.mainThread())
				.subscribe(new Consumer<AccountInfoResponse>() {
					@Override
					public void accept(@NonNull AccountInfoResponse accountInfoResponse) throws Exception {
						PutioAccountInfo account = accountInfoResponse.getInfo();
						PutioAccountInfo.DiskInfo disk = account.getDisk();

						textName.setText(account.getUsername());
						textEmail.setText(account.getMail());
						textDiskFree.setText(
								PutioUtils.humanReadableByteCount(disk.getAvail(), false));
						textDiskTotal.setText(getString(R.string.total_is,
								PutioUtils.humanReadableByteCount(disk.getSize(), false)));
					}
				}, new Consumer<Throwable>() {
					@Override
					public void accept(@NonNull Throwable throwable) throws Exception {
						PutioUtils.getRxJavaThrowable(throwable).printStackTrace();
					}
				});
	}
}

File: app/src/main/java/com/stevenschoen/putionew/account/AccountFragment.kt
package com.stevenschoen.putionew.account

import android.content.Intent
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.core.net.toUri
import com.stevenschoen.putionew.PutioUtils
import com.stevenschoen.putionew.R
import com.stevenschoen.putionew.putioApp
import com.trello.rxlifecycle3.components.support.RxFragment
import com.trello.rxlifecycle3.kotlin.bindToLifecycle
import io.reactivex.android.schedulers.AndroidSchedulers

class AccountFragment : RxFragment() {

  private var utils: PutioUtils? = null

  private lateinit var diskUsageView: TextView
  private lateinit var emailView: TextView
  private lateinit var usernameView: TextView
  private lateinit var manageOnWebView: View

  private lateinit var diskUsageDrawable: DiskUsageDrawable

  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)

    this.utils = putioApp.putioUtils
  }

  override fun onCreateView(
      inflater: LayoutInflater, container: ViewGroup?,
      savedInstanceState: Bundle?
  ): View {
    val view = inflater.inflate(R.layout.account, container, false)

    diskUsageView = view.findViewById(R.id.account_diskusage)
    usernameView = view.findViewById(R.id.account_username)
    emailView = view.findViewById(R.id.account_email)
    manageOnWebView = view.findViewById(R.id.account_manageonweb)

    diskUsageDrawable = DiskUsageDrawable(context!!)
    diskUsageView.background = diskUsageDrawable

    manageOnWebView.setOnClickListener {
      startActivity(Intent(Intent.ACTION_VIEW, "https://app.put.io/settings/".toUri()))
    }

    invalidateAccountInfo()

    return view
  }

  fun invalidateAccountInfo() {
    utils!!.restInterface.account()
        .bindToLifecycle(this)
        .observeOn(AndroidSchedulers.mainThread())
        .subscribe({ accountInfoResponse ->
          val account = accountInfoResponse.info
          val disk = account.disk

          usernameView.text = account.username
          emailView.text = account.mail
          diskUsageDrawable.usedFraction = disk.used.toFloat() / disk.size
          diskUsageView.text = getString(
              R.string.x_of_x_free,
              PutioUtils.humanReadableByteCount(disk.avail, false),
              PutioUtils.humanReadableByteCount(disk.size, false)
          )
        }, { throwable -> PutioUtils.getRxJavaThrowable(throwable).printStackTrace() })
  }
}
